generators/generator-base.js

```diff
@@ -1472,7 +1472,7 @@ module.exports = class extends PrivateBase {
      * @param {string} version - (optional) explicit dependency version number
      */
     addGradleDependencyInDirectory(directory, scope, group, name, version) {
-        const fullPath = `${directory}/build.gradle`;
+        const fullPath = `${directory.resolve(directory, 'build.gradle')}`;
```

Comment:
@BenEdridge This won't work, you need to use the `resolve` method from the [path module](https://nodejs.org/api/path.html#path_path_resolve_paths). Like:

```
const fullPath = path.resolve(directory, 'build.gradle'); 
```

Also, could you apply that to the `addMavenDependencyInDirectory` method?